File: 1128. Remove All Adjacent Duplicates In String/Solution03.cpp
/*
Submission Detail:{
    Difficulty : Easy
    Acceptance Rate : 70.58 %
    Runtime : 16 ms
    Memory Usage : 10.3 MB
    Testcase : 106 / 106 passed
    Ranking : 
        Your runtime beats 94.78 % of cpp submissions.
        Your memory usage beats 83.07 % of cpp submissions.
}
*/

class Solution {
public:
    string twoPointers(string &S) {
        int write_idx = 0, curr_idx = 0;
        
        while(curr_idx < S.size()) {
            // Last added char is a duplicate of current, so ignore both
            if(write_idx > 0 && S[write_idx-1] == S[curr_idx]) 
                write_idx -= 1;
            else
                S[write_idx++] = S[curr_idx];
            ++curr_idx;
        }
        
        return S.substr(0, write_idx);
    }
    
    string removeDuplicates(string S) {
        // return stackSol(S);
        return twoPointers(S);
    }
};

File: 0252. Meeting Rooms/Solution03.java
/*
Submission Detail:{
    Difficulty : Easy
    Acceptance Rate : 57.03 %
    Runtime : 10 ms
    Memory Usage : 45 MB
    Testcase : 78 / 78 passed
    Ranking : 
        Your runtime beats 63.08 % of java submissions.
        Your memory usage beats 49.81 % of java submissions.
}
*/

class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        
        
        boolean answer= true;
        Arrays.sort(intervals,(a, b)->{
            return a[0]-b[0];
            
        });
        
        //0,30  5,10 15,20
        for(int i=0; i<intervals.length-1;i++){
            if(intervals[i][1]>intervals[i+1][0]){
                return false;
            }
        }
        return answer;
        
    }
}